### PR TITLE
Migrate away from ILogger in encryption

### PR DESCRIPTION
**Pull request by come-nc**
Migrate away from ILogger in encryption and switch to constructor parameter promotion in concerned classes.

See #32127 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/encryption/lib/Crypto/Crypt.php [hunk 1]
```diff
@@ -593,10 +563,9 @@ private function splitMetaData($catFile, $cipher) {
 	 *
 	 * @param string $catFile
 	 * @param string $cipher
-	 * @return bool
 	 * @throws GenericEncryptionException
 	 */
-	private function hasSignature($catFile, $cipher) {
+	private function hasSignature($catFile, $cipher): bool {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:568]:
```suggestion
	private function hasSignature(string $catFile, string $cipher): bool {
```

File: apps/encryption/lib/Crypto/Crypt.php [hunk 2]
```diff
@@ -564,9 +535,8 @@ private function removePadding($padded, $hasSignature = false) {
 	 *
 	 * @param string $catFile
 	 * @param string $cipher
-	 * @return array
 	 */
-	private function splitMetaData($catFile, $cipher) {
+	private function splitMetaData($catFile, $cipher): array {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:539]:
```suggestion
	private function splitMetaData(string $catFile, string $cipher): array {
```

File: apps/encryption/lib/Crypto/Crypt.php [hunk 3]
```diff
@@ -172,10 +155,8 @@ public function getOpenSSLPKey() {
 
 	/**
 	 * get openSSL Config
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:157]: This comment is exactly the same as the function name, it's as good as removed.

**Review by nfebe — COMMENTED** [review]
Logger migration looks consistent to me, left inline comments about code comments and type hinting.

File: apps/encryption/lib/Crypto/Crypt.php [hunk 4]
```diff
@@ -240,10 +218,9 @@ public function generateHeader($keyFormat = self::DEFAULT_KEY_FORMAT) {
 	 * @param string $iv
 	 * @param string $passPhrase
 	 * @param string $cipher
-	 * @return string
 	 * @throws EncryptionFailedException
 	 */
-	private function encrypt($plainContent, $iv, $passPhrase = '', $cipher = self::DEFAULT_CIPHER) {
+	private function encrypt($plainContent, $iv, $passPhrase = '', $cipher = self::DEFAULT_CIPHER): string {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:223]: Sorry I did not mark all of them earlier, I assumed two mentions where enough.
```suggestion
	private function encrypt(string $plainContent, $iv, string $passPhrase = '', string $cipher = self::DEFAULT_CIPHER): string {
```

That might also lead to comment removal above.

File: apps/encryption/lib/Crypto/Crypt.php [hunk 5]
```diff
@@ -336,18 +311,16 @@ public function getLegacyCipher() {
 	/**
 	 * @param string $encryptedContent
 	 * @param string $iv
-	 * @return string
 	 */
-	private function concatIV($encryptedContent, $iv) {
+	private function concatIV($encryptedContent, $iv): string {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:315]:
```suggestion
	private function concatIV(string $encryptedContent, string $iv): string {
```

File: apps/encryption/lib/Crypto/Crypt.php [hunk 6]
```diff
@@ -336,18 +311,16 @@ public function getLegacyCipher() {
...
 		return $encryptedContent . '00iv00' . $iv;
 	}
 
 	/**
 	 * @param string $encryptedContent
 	 * @param string $signature
-	 * @return string
 	 */
-	private function concatSig($encryptedContent, $signature) {
+	private function concatSig($encryptedContent, $signature): string {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:323]:
```suggestion
	private function concatSig(string $encryptedContent, string $signature): string {
```

Comment (Fenn-CS) [reply apps/encryption/lib/Crypto/Crypt.php:323]: Function comment annotation might equally need removal

File: apps/encryption/lib/Crypto/Crypt.php [hunk 7]
```diff
@@ -357,19 +330,15 @@ private function concatSig($encryptedContent, $signature) {
 	 * encrypted content and is not used in any crypto primitive.
 	 *
 	 * @param string $data
-	 * @return string
 	 */
-	private function addPadding($data) {
+	private function addPadding($data): string {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:334]:
```suggestion
	private function addPadding(string $data): string {
```

File: apps/encryption/lib/Crypto/Crypt.php [hunk 8]
```diff
@@ -546,9 +515,8 @@ private function createSignature(string $data, string $passPhrase): string {
 	 *
 	 * @param string $padded
 	 * @param bool $hasSignature did the block contain a signature, in this case we use a different padding
-	 * @return string|false
 	 */
-	private function removePadding($padded, $hasSignature = false) {
+	private function removePadding($padded, $hasSignature = false): string|false {
```
Comment (Fenn-CS) [inline apps/encryption/lib/Crypto/Crypt.php:519]:
```suggestion
	private function removePadding(string $padded, bool $hasSignature = false): string|false {
```

**Review by nfebe — APPROVED** [review]
Sorry I did not mention all the hint improvements, I guess if you are feeling good you can amend the latest improvements to last commit.


Thanks!

File: apps/encryption/lib/Crypto/Encryption.php [hunk 9]
```diff
@@ -64,85 +59,44 @@ class Encryption implements IEncryptionModule {
 	/** @var string */
 	private $user;
 
-	/** @var  array */
-	private $owner;
+	private array $owner;
```
Comment (artonge) [inline apps/encryption/lib/Crypto/Encryption.php:62]: Why is it the only one getting a type?

**Review by artonge — APPROVED** [review]

**Pull request merged**